File: port_js/org/kbinani/vsq/IconHandle.js
/*
 * IconHandle.js
 * Copyright (C) 2008-2010 kbinani
 *
 * This file is part of org.kbinani.vsq.
 *
 * org.kbinani.vsq is free software; you can redistribute it and/or
 * modify it under the terms of the BSD License.
 *
 * org.kbinani.vsq is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 */
if( org == undefined ) var org = {};
if( org.kbinani == undefined ) org.kbinani = {};
if( org.kbinani.vsq == undefined ) org.kbinani.vsq = {};
if( org.kbinani.vsq.IconHandle == undefined ){

    /// <summary>
    /// 歌手設定を表します。
    /// </summary>

    /// <summary>
    /// 新しい歌手設定のインスタンスを初期化します。
    /// </summary>
    org.kbinani.vsq.IconHandle = function(){
        /// <summary>
        /// キャプション。
        /// </summary>
        this.Caption = "";
        /// <summary>
        /// この歌手設定を一意に識別するためのIDです。
        /// </summary>
        this.IconID = "";
        /// <summary>
        /// ユーザ・フレンドリー名。
        /// このフィールドの値は、他の歌手設定のユーザ・フレンドリー名と重複する場合があります。
        /// </summary>
        this.IDS = "";
        this.Index = 0;
        /// <summary>
        /// ゲートタイム長さ。
        /// </summary>
        this.Length = 0;
        this.Original = 0;
        this.Program = 0;
        this.Language = 0;
    };

    org.kbinani.vsq.IconHandle.ptototype = {
        /**
         * ゲートタイム長さを取得します。
         *
         * @return [int]
         */
        getLength : function() {
            return this.Length;
        },

        /**
         * ゲートタイム長さを設定します。
         *
         * @param value [int]
         * @return [void]
         */
        setLength : function( value ) {
            this.Length = value;
        },

        /**
         * このインスタンスと、指定された歌手変更のインスタンスが等しいかどうかを判定します。
         *
         * @param item [IconHandle] 比較対象の歌手変更。
         * @returns [bool] このインスタンスと、比較対象の歌手変更が等しければtrue、そうでなければfalseを返します。
         */
        equals : function( item ) {
            if ( item == null ) {
                return false;
            } else {
                return this.IconID === item.IconID;
            }
        },

        /**
         * このインスタンスのコピーを作成します。
         *
         * @return [object]
         */
        clone : function() {
            var ret = new org.kbinani.vsq.IconHandle();
            ret.Caption = this.Caption;
            ret.IconID = this.IconID;
            ret.IDS = this.IDS;
            ret.Index = this.Index;
            ret.Language = this.Language;
            ret.setLength( this.Length );
            ret.Original = this.Original;
            ret.Program = this.Program;
            return ret;
        },

        /**
         * この歌手設定のインスタンスを、VsqHandleに型キャストします。
         *
         * @return [VsqHandle]
         */
        castToVsqHandle : function() {
            var ret = new org.kbinani.vsq.VsqHandle();
            ret.m_type = org.kbinani.vsq.VsqHandleType.Singer;
            ret.Caption = this.Caption;
            ret.IconID = this.IconID;
            ret.IDS = this.IDS;
            ret.Index = this.Index;
            ret.Language = this.Language;
            ret.setLength( this.Length );
            ret.Program = this.Program;
            return ret;
        },
    };

}

File: src/cadencii.vsq/IconHandle.cs
/*
 * IconHandle.cs
 * Copyright © 2008-2011 kbinani
 *
 * This file is part of cadencii.vsq.
 *
 * cadencii.vsq is free software; you can redistribute it and/or
 * modify it under the terms of the BSD License.
 *
 * cadencii.vsq is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 */
using System;

namespace cadencii.vsq
{
    /// <summary>
    /// 歌手設定を表します。
    /// </summary>
    [Serializable]
    public class IconHandle : ICloneable
    {
        /// <summary>
        /// キャプション。
        /// </summary>
        public string Caption;
        /// <summary>
        /// この歌手設定を一意に識別するためのIDです。
        /// </summary>
        public string IconID;
        /// <summary>
        /// ユーザ・フレンドリー名。
        /// このフィールドの値は、他の歌手設定のユーザ・フレンドリー名と重複する場合があります。
        /// </summary>
        public string IDS;
        public int Index;
        /// <summary>
        /// ゲートタイム長さ。
        /// </summary>
        public int Length;
        public int Original;
        public int Program;
        public int Language;

        /// <summary>
        /// 新しい歌手設定のインスタンスを初期化します。
        /// </summary>
        public IconHandle()
        {
            Caption = "";
            IconID = "";
            IDS = "";
        }

        /// <summary>
        /// ゲートタイム長さを取得します。
        /// </summary>
        /// <returns></returns>
        public int getLength()
        {
            return Length;
        }

        /// <summary>
        /// ゲートタイム長さを設定します。
        /// </summary>
        /// <param name="value"></param>
        public void setLength(int value)
        {
            Length = value;
        }

        /// <summary>
        /// このインスタンスと、指定された歌手変更のインスタンスが等しいかどうかを判定します。
        /// </summary>
        /// <param name="item">比較対象の歌手変更。</param>
        /// <returns>このインスタンスと、比較対象の歌手変更が等しければtrue、そうでなければfalseを返します。</returns>
        public bool equals(IconHandle item)
        {
            if (item == null) {
                return false;
            } else {
                return IconID.Equals(item.IconID);
            }
        }

        /// <summary>
        /// このインスタンスのコピーを作成します。
        /// </summary>
        /// <returns></returns>
        public Object clone()
        {
            IconHandle ret = new IconHandle();
            ret.Caption = Caption;
            ret.IconID = IconID;
            ret.IDS = IDS;
            ret.Index = Index;
            ret.Language = Language;
            ret.setLength(Length);
            ret.Original = Original;
            ret.Program = Program;
            return ret;
        }

        /// <summary>
        /// このインスタンスのコピーを作成します。
        /// </summary>
        /// <returns></returns>
        public object Clone()
        {
            return clone();
        }

    }

}
